src/main/java/com/hubspot/jackson/datatype/protobuf/builtin/serializers/MessageSerializer.java

```diff
@@ -117,7 +117,39 @@ public boolean isUnwrappingSerializer() {
 
   @Override
   public MessageSerializer unwrappingSerializer(NameTransformer nameTransformer) {
-    return new MessageSerializer(config, true);
+    return new MessageSerializer(getConfig(), true);
+  }
+
+  @Override
+  public void acceptJsonFormatVisitor(JsonFormatVisitorWrapper visitor, JavaType typeHint) throws JsonMappingException {
+    Message defaultInstance = defaultInstance(typeHint);
+    Descriptor descriptor = defaultInstance.getDescriptorForType();
+    Function<FieldDescriptor, String> propertyNaming = getPropertyNaming(descriptor, visitor.getProvider());
+
+    if (isUnwrappingSerializer()) {
+      // TODO do something different?
+    }
```

Comment:
I need to do some testing to see how `@JsonUnwrapped` is supposed to interact with schema generation

Reply:
Seems like it "just works"